src/Fees.sol

```diff
@@ -52,44 +50,22 @@ abstract contract Fees is IFees, Owned {
             assembly {
                 returnData := mload(add(_data, 0x20))
             }
-            // Ensure return data does not overflow a uint24 and that the underlying fees are within bounds.
-            (success, protocolFees) = returnData == uint24(returnData) && _isValidProtocolFees(uint24(returnData))
-                ? (true, uint24(returnData))
+            // Ensure return data does not overflow a uint16 and that the underlying fees are within bounds.
+            (success, protocolFees) = returnData == uint16(returnData) && _isFeeWithinBounds(uint16(returnData))
+                ? (true, uint16(returnData))
                 : (false, 0);
         }
     }
 
-    /// @notice There is no cap on the hook fee, but it is specified as a percentage taken on the amount after the protocol fee is applied, if there is a protocol fee.
-    function _fetchHookFees(PoolKey memory key) internal view returns (uint24 hookFees) {
-        if (address(key.hooks) != address(0)) {
-            try IHookFeeManager(address(key.hooks)).getHookFees(key) returns (uint24 hookFeesRaw) {
-                uint24 swapFeeMask = key.fee.hasHookSwapFee() ? 0xFFF000 : 0;
-                uint24 withdrawFeeMask = key.fee.hasHookWithdrawFee() ? 0xFFF : 0;
-                uint24 fullFeeMask = swapFeeMask | withdrawFeeMask;
-                hookFees = hookFeesRaw & fullFeeMask;
-            } catch {}
-        }
-    }
-
     function _fetchDynamicSwapFee(PoolKey memory key) internal view returns (uint24 dynamicSwapFee) {
         dynamicSwapFee = IDynamicFeeManager(address(key.hooks)).getFee(msg.sender, key);
         if (dynamicSwapFee >= MAX_SWAP_FEE) revert FeeTooLarge();
     }
 
-    function _isValidProtocolFees(uint24 protocolFees) internal pure returns (bool) {
-        if (protocolFees != 0) {
-            uint16 protocolSwapFee = uint16(protocolFees >> 12);
-            uint16 protocolWithdrawFee = uint16(protocolFees & 0xFFF);
-            return _isFeeWithinBounds(protocolSwapFee) && _isFeeWithinBounds(protocolWithdrawFee);
-        }
-        return true;
-    }
-
-    /// @dev Only the lower 12 bits are used here to encode the fee denominator.
     function _isFeeWithinBounds(uint16 fee) internal pure returns (bool) {
         if (fee != 0) {
-            uint16 fee0 = fee % 64;
-            uint16 fee1 = fee >> 6;
+            uint16 fee0 = fee % 256;
```

Comment:
Why change this? If we want to increase protocol fee granularity I think thats a separate discussion. I think this means the fee can be a lot smaller and I thought we intentionally had it bound before (both from a min and a max).

In any case it's worth a comment on what the min fee value is now that there are 8 bits that can be used instead of 6.

Reply:
Yeah I think that's fine, can you add a comment? I just wasn't expecting that to change, not that its a bad change but just deserves some reasoning :)